bc_obps/reporting/service/report_activity_save_service.py

```diff
@@ -81,13 +80,18 @@ def save(self, data: dict) -> ReportActivity:
             defaults={"json_data": activity_data},
         )
 
-        # Delete the existing report_source_types with an id not in the form_data (this means they've been deleted on the form)
-        ReportSourceType.objects.filter(report_activity=report_activity).exclude(
-            id__in=retrieve_ids(data["sourceTypes"])
-        ).delete()
+        # Check if "sourceTypes" exists in the data before processing
+        if "sourceTypes" in data:
+            # Delete the existing report_source_types with an id not in the form_data (this means they've been deleted on the form)
+            ReportSourceType.objects.filter(report_activity=report_activity).exclude(
+                id__in=retrieve_ids(data["sourceTypes"])
+            ).delete()
 
-        for source_type_key in data["sourceTypes"]:
-            self.save_source_type(report_activity, source_type_key, data["sourceTypes"][source_type_key])
+            for source_type_key in data["sourceTypes"]:
+                self.save_source_type(report_activity, source_type_key, data["sourceTypes"][source_type_key])
+
+        # Save raw data after processing source types
+        self.save_raw_data(data)
```

Comment:
I think you're right, the save method modifies in-place the json data. This is not best practice, but works well for this particular case.
We also need a test to validate this new behaviour